src/routing/approved.rs

```diff
@@ -1451,6 +1465,9 @@ impl Approved {
 
         if !new_is_elder {
             commands.extend(self.return_relocate_promise());
+        } else if &old_elders_info != self.section.elders_info() {
+            // Whenever there is an elders change, casting a round of join_flag vote to sync.
+            commands.extend(self.vote(Vote::JoinFlag(self.join_flag))?);
```

Comment:
So it seems to me that either the join flag should not be voted for at all or if it is, it should be part of the `Section` state in which case it would get synced via the `Sync` message already. In either case, we probalby don't need this.

Reply:
maybe should be part of the section. I will think about it.